controllers/argocd/deployment.go

```diff
@@ -350,6 +343,59 @@ func isMergable(extraArgs []string, cmd []string) error {
 	return nil
 }
 
+// appendUniqueArgs appends extraArgs to cmd while ignoring any duplicate flags.
+func appendUniqueArgs(cmd []string, extraArgs []string) []string {
+	// Parse cmd into a map to track flags and their indices
+	existingArgs := make(map[string]int) // Map to track index of each flag in cmd
```

Comment:
Based on our Slack discussion, could you try using a `map[string]string` to simplify the logic?

Reply:
@svghadi I tried updating the logic to use map[string]string but found issues as mentioned-
- map[string]string creates a map of flags to values, meaning it's expecting each flag to be associated with a value (or an empty string if no value is provided) this approach has the risk of treating arguments like --staticassets as executable commands if there is an issue in the flag parsing. If the function expects this argument to be passed as part of a command and it's incorrectly handled (i.e., not paired with a valid value), this could lead to the error like - create failed: unable to start container process: exec: \"--staticassets\": executable file not found in $PATH"

- On the other hand, map[string]int  keeps track of the exact positions of flags in the original cmd array, so any updates or additions to cmd are carefully done by index. This prevents the accidental misinterpretation of arguments and ensures the correct structure for the final command.

I may be mistaken, but PTAL and let me know your thoughts. I'm open to any suggestions